File: WebSharper.Templates/templates/Service-CSharp/Service.cs
using WebSharper;
using WebSharper.Sitelets;

public class Service
{
    [EndPoint("/")]
    public record Home;

    // sample endpoint: /user/1
    [EndPoint("GET /user/{Id}")]
    public record GetUser(int Id);

    public record User(int Id, string Name);

    [Website]
    public static Sitelet<object> Main =>
        new SiteletBuilder()
            .With<Home>((ctx, _) => Content.Text("Service version 1.0"))
            .WithCors<GetUser>(
                corsBuilder => corsBuilder.WithOrigins("http://example.com"),
                (ctx, action) => Content.Json(new User(action.Id, "John"))
            )
            .Install();
}

File: WebSharper.Templates/templates/Service-FSharp/Service.fs
module Service

open WebSharper
open WebSharper.Sitelets

type EndPointWithCors =
    | [<EndPoint "GET /user">] GetUser of Id: int 

type EndPoint =
    | [<EndPoint "/">] Home
    | [<EndPoint "/">] EndPointWithCors of Cors<EndPointWithCors>

type User =
    { 
        Id: int
        Name: string
    }

let HandleApi ctx endpoint =
    match endpoint with
    | GetUser uid ->
        Content.Json { Id = uid; Name = "John" }

[<Website>]
let Main =
    Application.MultiPage (fun ctx endpoint ->
        match endpoint with
        | Home -> Content.Text "Service version 1.0"
        | EndPointWithCors endpoint ->
            Content.Cors endpoint 
                (fun corsAllows ->
                    { corsAllows with
                        Origins = ["http://example.com"]
                    }
                )
                (HandleApi ctx)
    )

